lib/vssolution.cpp

```diff
@@ -727,7 +727,44 @@ int VisualizationSceneSolution::GetRefinedValuesAndNormals(
       // In 1D we do not have well-defined normals.
       if (dim > 1)
       {
-         rsol->GetGradients(i, ir, tr);
+         if (rsol->FESpace()->GetFE(i)->GetMapType() == FiniteElement::VALUE)
+         {
+            rsol->GetGradients(i, ir, tr);
+         }
+         else
```

Comment:
I think we should be explicit here (i.e. `FiniteElement::INTEGRAL`) with the mapping type selection and reserve the else branch for error handling (e.g. UNKNOWN_MAP_TYPE ).

Reply:
Done 👍